ompi/mca/coll/ucc/coll_ucc_module.c

```diff
@@ -175,10 +175,16 @@ static ucc_status_t oob_allgather_test(void *req)
         senddatafrom = (rank - oob_req->iter + size) % size;
         tmprecv = (char*)oob_req->rbuf + (ptrdiff_t)recvdatafrom * (ptrdiff_t)msglen;
         tmpsend = (char*)oob_req->rbuf + (ptrdiff_t)senddatafrom * (ptrdiff_t)msglen;
-        MCA_PML_CALL(isend(tmpsend, msglen, MPI_BYTE, sendto, MCA_COLL_BASE_TAG_UCC,
+        rc = MCA_PML_CALL(isend(tmpsend, msglen, MPI_BYTE, sendto, MCA_COLL_BASE_TAG_UCC,
                            MCA_PML_BASE_SEND_STANDARD, comm, &oob_req->reqs[0]));
-        MCA_PML_CALL(irecv(tmprecv, msglen, MPI_BYTE, recvfrom,
+	if (OMPI_SUCCESS != rc) {
+            return rc;
```

Comment:
I'm not sure if conversion is safe here, return value of this function is ucc_status_t and might be not compatible with ompi error codes.  If rc is positive number then it's actually not an error from ucc perspective since all errors in ucc are negative. Maybe just to make it safe the function returns UCC_ERR_NO_MESSAGE in case of error.

Reply:
I don't think this is correct. The macro ends up calling the `.pml_isend` field in the PML struct, which in the case of UCX points to [mca_pml_ucx_isend](https://github.com/open-mpi/ompi/blob/08ac1bec07caefae564160ef61bb3850d047f8d5/ompi/mca/pml/ucx/pml_ucx.c#L890), a function that returns OMPI errors.